packages/projection-typeorm/src/entity/BlockData.entity.ts

```diff
@@ -0,0 +1,17 @@
+import { BlockCascadeRelationOptions, BlockEntity } from './Block.entity';
+import { Cardano } from '@cardano-sdk/core';
+import { Column, Entity, JoinColumn, OneToOne, PrimaryGeneratedColumn } from 'typeorm';
+import { json, serializableObj, stringBytea } from './transformers';
+
+@Entity('block_data')
+export class BlockDataEntity {
+  @PrimaryGeneratedColumn()
+  id?: number;
+
+  @OneToOne(() => BlockEntity, BlockCascadeRelationOptions)
+  @JoinColumn()
+  block?: BlockEntity;
+
+  @Column({ transformer: [serializableObj, json, stringBytea], type: 'bytea' })
```

Comment:
The advantages of a byte array Vs a string type are:

1. the ability to write non printable characters
2. avoiding some escape/un-escape of single quote character

the main disadvantage is: while developing/debugging, querying a byte array from DB results in an opaque string.

Using a byte array field to store a JSON encoded string has two great disadvantages:

1. it uses a lot of space
2. it produces an opaque string

If we want to settle on an opaque string (less handy while developing/debugging) I think it would be better to optimize the data space: CBOR serialization produces a binary tiny string which perfectly fit into a bytes array field.

On the other hand, we can renounce to heavy space optimization to gain the benefit of a human readable string. In this case a string field rather than a bytes array string is required.

Reply:
Hm, this is not supposed to be queryable and needs to be optimized for disk space.

> it uses a lot of space

Are you sure? I chose `bytea` because it should take less space :sweat_smile: Need to do some additional research...

Reply:
I'm not so prepared on the headers of the two field types, but I thing it's negligible if compared to the size of a block.
I would focus on the content of the field: if you think we will no need to query for its content while developing debugging I would say this is enough to opt for a CBOR serialization.

Reply:
Note it's not CBOR that we're saving, it's JSON. In the future it's likely we'd change it to CBOR.

We could create a SPIKE for this, we're free to change it until the service is deployed.